### ISSUE TITLE
Dev toolbar axe app hidden by popups and hightlights

### ISSUE THREAD

**Issue by OliverSpeir** [withastro/astro#11114]
### Describe the Bug

Accessibility toolbar can be rendered unusable due to overlapping overlays and hover tips

Ironically it can actually create an accessibility issue, when there are many overlapping highlight overlays the contrast ratio of the text in the main window drops.
<img width="784" alt="Screenshot 2024-05-22 at 1 07 04 AM" src="https://github.com/withastro/astro/assets/115520730/81755a6b-d28a-40aa-bd3b-880ae1abb6de">
<img width="1040" alt="Screenshot 2024-05-22 at 12 59 04 AM" src="https://github.com/withastro/astro/assets/115520730/9e0a2c44-d1f0-4ae8-944c-cca5451101eb">

### What's the expected result?

Not sure exactly, but I think the main window should always be on top and maybe when one of the list items is selected all are highlights are hidden 

### Link to Minimal Reproducible Example

Can be reproduced with a fresh starlight install

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by AdamGEmerson** [comment]
I'm going to try taking a look at this.

**Comment by ematipico** [comment]
Thank you @AdamGEmerson , I assigned the issue to you

**Comment by ematipico** [comment]
@AdamGEmerson any updates?

**Comment by katungi** [comment]
Hey @ematipico , this issue looks dormant, can I take a look at it?

**Comment by ematipico** [comment]
Sure, it's yours

**Comment by AdamGEmerson** [comment]
Yeah, sorry about that.  I was able to replicate the issue but got caught up with other things and lost momentum.

**Comment by katungi** [comment]
Hey @AdamGEmerson , no problem, we can always collaborate on it. if you are on discord, we can chat more about it - here's my username, `warriorrapid`

**Comment by bestabyte** [comment]
Hi @ematipico
Can I take a look this?

**Comment by ematipico** [comment]
It's yours @bestabyte

**Comment by ematipico** [comment]
Assigned to you @jpwienekus
